adr/adr-1.md

```diff
@@ -0,0 +1,124 @@
+
+The native executable agent is responsible for the life cycle of all child processes, whether they are wrapped in a microVM or not. It must be able to gracefully terminate all of those processes on request.
+
+The template used for building the on-demand root file system is configurable in the appropriate workload type section of the node configuration.
+
+The native agent is responsible for supplying the executable workload with enough information to connect to host services. Where the function-type agents expose a host services proxy, the native workloads are allowed to connect directly to the indicated URL/creds.
+
+### JavaScript Agent
+The JavaScript agent is responsible for executing individual _functions_. These functions are designed to be short-lived and the agent can choose to reclaim resources from those functions whenever it deems necessary.
+
+The agent can execute a single JavaScript file (`.js`) or it can take an archive file as the artifact, where it will extract all `.js` files from the archive and run them in a single context, provided there remains only one exported `run` function.
+
+The JavaScript agent is responsible for exposing host services to each function. The agent must also ensure that no two workloads share the same execution context.
+
+### WebAssembly Agent
+The WebAssembly agent is responsible for executing WebAssembly _components_. We no longer support running the classic modules. All executable WebAssembly artifacts will be components with their interfaces defined using the `wit` convention (and likely the appropriate `wit-bindgen` tooling).
+
+The WebAssembly agent is responsible for exposing host services to each function, proxying access to the host services API exposed on the internal NATS server.
+
+### OCI Agent
+The OCI agent will be responsible for running images from an indicated container registry. This effectively gives Nex nodes the ability to schedule and start docker images.
+
+Information on how to connect to the appropriate host services URL/creds will be provided to the workload. OCI workloads are considered long-lived services and as such do not require a host services proxy.
+
+## New Agent Protocol
+Every agent is required to adhere to this agent protocol. The agents _must_ connect to the internal NATS server embedded in the Nex node host and this is the _only_ way that agents and Nex nodes communicate with each other other than when the Nex node initially spawns an agent.
+
+Note that in the table below, _direction_ is relative to the _agent_. `hostint.>` is an export from the Nex node account, while `agentint.>` is an export from the agent's user/account. All communications in the table below happens on the internal connection, _agents should not communicate with any NATS servers other than the Nex node host_.
```

Comment:
Given the connection is internal, is it necessary to have the `int` suffix in the subject? Will there be non-internal subjects?

Reply:
this is just a leftover from the status quo. We can get rid of the `int`.